reactos/dll/win32/browseui/basebarsite.cpp

```diff
@@ -410,14 +588,79 @@ HRESULT STDMETHODCALLTYPE CBaseBarSite::SetBandState(DWORD dwBandID, DWORD dwMas
 
 HRESULT STDMETHODCALLTYPE CBaseBarSite::RemoveBand(DWORD dwBandID)
 {
-    return E_NOTIMPL;
+    REBARBANDINFO                   bandInfo;
+    HRESULT                         hr;
+    CBarInfo                        *pInfo;
+    CComPtr<IObjectWithSite>        pSite;
+    CComPtr<IDeskBand>              pDockWnd;
+    DWORD                           index;
+
+    // Retrieve the right index of the coolbar knowing the id
+    index = SendMessage(RB_IDTOINDEX, dwBandID, 0);
+    if (index == 0xffffffff)
+       return E_INVALIDARG;
+
+    if (FAILED_UNEXPECTEDLY(GetInternalBandInfo(index, &bandInfo)))
+        return E_INVALIDARG;
+
+    pInfo = reinterpret_cast<CBarInfo*>(bandInfo.lParam);
+    if (!pInfo)
+        return E_INVALIDARG;
+
+    hr = pInfo->fTheBar->QueryInterface(IID_PPV_ARG(IDeskBand, &pDockWnd));
```

Comment:
Unless someone injected a rogue band with an invalid lParam field (or some corruption inside Wine's comctl32), it should be valid yes.